### PR TITLE
[CI:DOCS] Add socket_activation.md

### LINKED ISSUE #14908: Create a Podman socket activation tutorial
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

Add a Podman tutorial that describes Podman socket activation

* mention the two ways Podman supports socket activation: 
  + Podman API service
  + socket activation of containers
* provide some examples for running a container that is socket-acitvated
* mention that socket-activated traffic does not pass through slirp4netns.
  This gives different network performance characteristics (native speed but delayed start as the container has to start up)

### PR DESCRIPTION
**Pull request by eriksjolund**
Closes #14908

Signed-off-by: Erik Sjölund <erik.sjolund@gmail.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
Nice work, @eriksjolund!

@rhatdan @giuseppe @TomSweeneyRedHat PTAL

**Comment by eriksjolund** [comment]
A side-note:

@rhatdan and I was [discussing](https://github.com/containers/podman/discussions/13390#discussioncomment-2858188) the possibility create a related blog post for
https://www.redhat.com/sysadmin/topics/containers

Here is my initial attempt:
https://github.com/eriksjolund/podman-socket-activation/blob/rework_text/blog.md
(For me it would be fine to be a co-author too if you want to modify it)

I also wrote another text
https://github.com/eriksjolund/podman-socket-activation/blob/rework_text/RestrictAddressFamilies.md
(that might be too avant garde for a redhat blog or the podman web page)

File: docs/tutorials/podman-socket-activation.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+through slirp4netns and gets the performance penalty that comes with it.
+
+Unfortunately, using _socket activation_ also comes with a disadvantage. The very first connection
+to a socket-activated container will have more latency due to container startup.
+To minimize this latency, consider adding the __podman run__ option __--pull=never__ and
+instead pull the container image beforehand.
+
+### Note about SELinux
+
+> **Note**
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/podman-socket-activation.md:199]: I'm not sure I like how the "Note" resolves.  I think I'd drop it.

**Comment by TomSweeneyRedHat** [comment]
@eriksjolund you've been rocking the tutorials of late.  Can you also please update the README.md with links to the new tutorials and a small description of each?

File: docs/tutorials/podman-socket-activation.md (hunk 1 above)
Comment (eriksjolund) [reply docs/tutorials/podman-socket-activation.md]: I removed both notes. The importance of the first note will eventually fade away as people are using newer versions of container-selinux.

The second note is probably more of interest to the end users that are super interested and not to the general end user. 

```
### Note about SELinux

> **Note**
> If your computer is running SELinux, you need to have [__container-selinux 2.186.0__](https://github.com/containers/container-selinux)
> or newer installed. If container socket activation via Podman does not work and you are using an older version of
> container-selinux, add `--security-opt label=disable` to `podman run` as a work around.

### Note about running systemd in container

> **Note** There is currently no way to pass a socket-activated socket into a container when the container is running
> systemd. In other words, the executable __/usr/lib/systemd/systemd__ does not support socket activation.
> (There is a feature request related to it https://github.com/systemd/systemd/issues/17764)
```

**Comment by eriksjolund** [comment]
> @eriksjolund you've been rocking the tutorials of late. Can you also please update the README.md with links to the new tutorials and a small description of each?

I added a link to the socket activation tutorial. (I'll wait with adding a link to the performance tutorial because I don't know how long it will take to finish it)


Another thing. I modified this PR to build a local HTTPD container instead of  of using _ghcr.io/eriksjolund/socket-activate-httpd_. The reason is that I wanted to avoid the responsibility to keep that container image up to date to the latest security fixes in HTTPD.

**Comment by eriksjolund** [comment]
First thought I should write something about the need to use the Linux operating system to be able to use socket activation.
I had difficulties coming up with a text about that  so I just skipped that part.
Maybe it's possible to use Windows or Mac with a Linux VM that use VSOCK?
I'm a little confused about the topic so I skipped writing about that too.

I remove the draft status.

**Comment by rhatdan** [comment]
/approve
LGTM
@eriksjolund Great work.  Did you ever publish the full blogs?
We can wait for @vrothberg to merge.

**Review by vrothberg — COMMENTED** [review]
/lgtm

**Comment by eriksjolund** [comment]
> @eriksjolund Great work. Did you ever publish the full blogs?

@rhatdan Yesterday I split out the tutorial about using the systemd directive [__RestrictAddressFamilies__](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RestrictAddressFamilies=)
into the new GitHub repository https://github.com/eriksjolund/podman-RestrictAddressFamilies
On Monday or Tuesday I plan to write to tweet about it and also about the new [Podman socket activation tutorial](https://github.com/containers/podman/blob/main/docs/tutorials/socket_activation.md)

I rewrote the [potential RedHat blog](https://github.com/eriksjolund/podman-socket-activation/blob/rework_text/blog.md) a bit. (I removed the SELinux part and instead discussed privileges in more general terms).
Do you think the blog could be published under https://www.redhat.com/sysadmin/topics/containers ?
If so, how should I proceed?

**Pull request merged**